src/server.cpp

```diff
@@ -20,19 +22,50 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
     if (!error || error == boost::asio::error::message_size)
     {
-        const std::string message("Hello, World!");
+        size_t length = recv_buffer_[2] * 256 + recv_buffer_[3];
+        if (bytes >= length)
+        {
+            uint8_t send_code;
 
-        socket_.async_send_to(boost::asio::buffer(message), remote_endpoint_,
-          std::bind(&Server::handle_send, this, message,pls::_1, pls::_2));
+            if (recv_buffer_[0] == ACCESS_REQUEST)
+                send_code = 2;
 
-        start_receive();
+            std::array<uint8_t, 4096> send_buffer;
+            uint8_t send_length1 = 0;
+            uint8_t send_length2 = 20;
+
+            send_buffer[0] = send_code;
+            send_buffer[1] = recv_buffer_[1];
+            send_buffer[2] = send_length1;
+            send_buffer[3] = send_length2;
+
+            for (int i = 4; i <= 19; i++)
+                send_buffer[i] = recv_buffer_[i];
+
+            std::string secr("secret");
+
+            for (int i = 0; i <= 5; i++)
+                send_buffer[i + 20] = secr[i];
+
+            unsigned char md[MD5_DIGEST_LENGTH];
+
+            MD5(send_buffer.data(), send_length2 + secr.length(), md);
+
+            for (int i = 0; i <= 15; i++)
+                send_buffer[i + 4] = md[i];
+
+          socket_.async_send_to(boost::asio::buffer(send_buffer), remote_endpoint_,
```

Comment:
`send_buffer` is a local variable. When `ioc` starts actual sending, it will be well out of scope and destroyed. You need to make it permanent. Make it a class member, for example.